modular_skyrat/modules/cyborg/code/robot_upgrades.dm

```diff
@@ -244,3 +244,41 @@
 		var/obj/item/inducer/cyborg/inducer = locate() in target_robot.model
 		if (inducer)
 			target_robot.model.remove_module(inducer, TRUE)
+
+// Borg picket sign upgrade
+/obj/item/borg/upgrade/sign
+	name = "Cyborg sign waving upgrade"
+	desc = "A sign unit for cyborgs, allowing them to easily advertise products through an intuitive, wave-able sign."
+	icon_state = "cyborg_upgrade3"
+	require_model = FALSE
+
+/obj/item/borg/upgrade/sign/action(mob/living/silicon/robot/target_robot, user = usr)
+	. = ..()
+	if(!.)
+		return
+	var/obj/item/picket_sign/cyborg/borgsign = locate() in target_robot.model.modules
+	if(borgsign)
+		target_robot.balloon_alert(user, "already equipped!")
+		return
+
+	borgsign = new(target_robot.model)
+	target_robot.model.basic_modules += borgsign
+	target_robot.model.add_module(borgsign, FALSE, TRUE)
+
+/obj/item/borg/upgrade/borgsign/deactivate(mob/living/silicon/robot/target_robot, user = usr)
+	. = ..()
+	if (!.)
+		return
+	var/obj/item/picket_sign/cyborg/borgsign = locate() in target_robot.model
+	if (borgsign)
+		target_robot.model.remove_module(borgsign, TRUE)
+
+///Retext proc for the picket sign that checks for a cyborg user instead of writing_instrument.
+/obj/item/picket_sign/cyborg/proc/borgsign_retext(mob/user)
+	if(!iscyborg(user))
+		return
+	var/txt = tgui_input_text(user, "What would you like to write on the sign?", "Sign Label", max_length = 30)
```

Comment:
Probably check for the text being the same as what the sign is already set to

Reply:
Yeah, you also need to sanitize the text here.
Currently it allows html injection

Reply:
@Helios7-1

Reply:
I feel like TGUI inputs already sanitize the input, don't they?